### ISSUE TITLE
What's the disk size should I have if run on bsc mainnet?

### ISSUE THREAD

**Issue by scream7** [binance-chain/bsc-erigon#10]
I have a node with 2TB ssd, but the sync progress is about 10m/17m when I run out of disk space.
So any recommendation? 4TB or above? thanks a lot.

**Comment by g4bwy** [comment]
Hi,

I finished syncing up to block 13749154, and without --prune= options (apparently pruning is disabled by default) the chaindata/mdbx.data file takes a little bit more than 4TB (4329327034368 bytes to be exact). I haven't tried enabling pruning yet so maybe it'll get smaller after that.

Be sure that you have enough SSD space and that your filesystem supports files larger than 4TB (I got caught the first time with that limit and had to reformat to a filesystem with larger files support)

**Comment by radian-rgc** [comment]
> I finished syncing up to block 13749154, and without --prune= options (apparently pruning is disabled by default) the chaindata/mdbx.data file takes a little bit more than 4TB (4329327034368 bytes to be exact). I haven't tried enabling pruning yet so maybe it'll get smaller after that.

Full Archive, just ~4TB ?

**Comment by g4bwy** [comment]
yes, that's a full archive. now 4.1TB at block 13842577

**Comment by radian-rgc** [comment]
Intresting.
May i know what is axactly your command and prune info? 
Thanks

**Comment by g4bwy** [comment]
I just used ```erigon --chain bsc --datadir /path/to/data``` + some --staticpeers / --bootnodes lists taken from bsc-geth's official config.toml. Also I'm running cmd/sentry on another machine to circumvent the low peers count issue I currently have on debian .

by default pruning is disabled, you'd have to add --prune=hrtc to enable it. by default it prunes state for all blocks older than the latest 90000. this can be set independently for each type of data to prune, ie.:
```
   --prune value                      Choose which ancient data delete from DB:
                                      h - prune history (ChangeSets, HistoryIndices - used by historical state access)
                                      r - prune receipts (Receipts, Logs, LogTopicIndex, LogAddressIndex - used by eth_getLogs and simil
ar RPC methods)
                                      t - prune transaction by it's hash index
                                      c - prune call traces (used by trace_* methods)
                                      Does delete data older than 90K block (can set another value by '--prune.*.older' flags).
                                      If item is NOT in the list - means NO pruning for this data.s
                                      Example: --prune=hrtc (default: "disabled")
   --prune.h.older value              Prune data after this amount of blocks (if --prune flag has 'h', then default is 90K) (default: 0)
   --prune.r.older value              Prune data after this amount of blocks (if --prune flag has 'r', then default is 90K) (default: 0)
   --prune.t.older value              Prune data after this amount of blocks (if --prune flag has 't', then default is 90K) (default: 0)
   --prune.c.older value              Prune data after this amount of blocks (if --prune flag has 'c', then default is 90K) (default: 0)
   --prune.h.before value             Prune data before this block (default: 0)
   --prune.r.before value             Prune data before this block (default: 0)
   --prune.t.before value             Prune data before this block (default: 0)
   --prune.c.before value             Prune data before this block (default: 0)
```

I just started syncing in a new datadir with pruning enabled (unfortunately, it doesn't seem possible to enable pruning on an existing full archive datadir), so I'll know in a few days before I can get an idea of the SSD space it takes (execution stage took around one week on my setup to complete, the whole process took about 10 days).

**Comment by radian-rgc** [comment]
Thanks @g4bwy 

If i want to just keep last 3mil archive, i must use this command right ? :

`./build/bin/erigon --chain bsc --prune.h.older=3000000 --prune.r.older=3000000 --prune.t.older=3000000 --prune.c.older=3000000 --datadir dexpool`

**Comment by g4bwy** [comment]
> Thanks @g4bwy
> 
> If i want to just keep last 3mil archive, i must use this command right ? :
> 
> `./build/bin/erigon --chain bsc --prune.h.older=3000000 --prune.r.older=3000000 --prune.t.older=3000000 --prune.c.older=3000000 --datadir dexpool`

you should also add ```---prune=hrtc```

**Comment by radian-rgc** [comment]
Thank you @g4bwy 

can you please share your telegram id ? Please i want your snapshot . i can pay for it

Please sir...

This is my tg: @radian_c

**Comment by g4bwy** [comment]
Hi,

Sorry, I had to ditch that archive snapshot, since apparently some of the manual hacks I did to overcome the errors caused by my previous use of a filesystem with 4TB file limit seem to have corrupted the database somehow.

I strongly suspect this was the cause of my other opened issue (#11) that made other peers drop my node due to some wrong state data sent in p2p handshakes.

Since restarting from scratch with pruning enabled, I'm running with ~95-99 peers without problems.

**Comment by g4bwy** [comment]
finished syncing with 90k pruning enabled a couple days ago, now synced at block 14154704, and the chaindata directory is at 1.4TB (1445257523250 bytes)

**Comment by radian-rgc** [comment]
@g4bwy 

Cool,

May i know what exactly your hardware information? And from where hosted ?

**Comment by g4bwy** [comment]
> May i know what exactly your hardware information? And from where hosted ?

Hetzner AX101 (Ryzen 9 5950X 16 Core / 128GB RAM / 2x3.8TB NVMe, raid-0 striping, XFS format)

**Comment by radian-rgc** [comment]
> > May i know what exactly your hardware information? And from where hosted ?
> 
> Hetzner AX101 (Ryzen 9 5950X 16 Core / 128GB RAM / 2x3.8TB NVMe, raid-0 striping, XFS format)

Exactly like my server.

Now im in 6/16 stage and almost 9.700 blocks synced (after 5days) and fetched ~20block per second.

Is it normal?

**Comment by nikolinsko** [comment]
Hello,

can anyone shed some light on the space requirements in the initial sync phase with pruning enabled? Will blocks be pruned during this phase or do you need the 4.1TB regardless of pruning to complete the sync?

 I'm planning to rent a rather cheap server at hetzner with 2x1T m2 ssd which will be setup in raid0. I'm also a bit worried that the CPU might be bottlenecked (i7 8700) but the tests made by ankr suggest that this should not be an issue?

**Comment by g4bwy** [comment]
> Hello,
> 
> can anyone shed some light on the space requirements in the initial sync phase with pruning enabled? Will blocks be pruned during this phase or do you need the 4.1TB regardless of pruning to complete the sync?

No, to complete a full sync with pruning you don't need 4TB. most of the disk space is filled in 2 phases: block headers/body download + storage (which currently takes up ~1.2TB, which is basically the size of the raw chain), and state generation through block execution (stage 6/16), which generates the biggest part of the 4TB when pruning is disabled.

When you enable pruning, stage 6/16 prunes the states during execution, so it doesn't fill up as much.

**Comment by g4bwy** [comment]
> I'm planning to rent a rather cheap server at hetzner with 2x1T m2 ssd which will be setup in raid0. I'm also a bit worried that the CPU might be bottlenecked (i7 8700) but the tests made by ankr suggest that this should not be an issue?

about the CPU, it might be a good idea to perform the execution (6/16) stage of the initial sync on a bigger machine, then copy your db to a cheaper one, since this is really the longest stage, and it requires a lot of single-thread CPU performance.

**Comment by nikolinsko** [comment]
@g4bwy thanks alot. I'm not that worried about the delay caused by the aged cpu, i guess i'll just give it a go :)

Another thing, has anyone tried out the official ankr bsc branch: https://github.com/Ankr-network/erigon, seems like this repo was "forked" from it and it has more recent commits.

**Comment by g4bwy** [comment]
Well, it certainly seems whoever from the bsc team forked this repo isn't actively pursuing development at this time (at least publicly).

All issues opened here appear to just be messages in a bottle, so maybe we should either move discussion to ankr's repo (or even ledgerwatch's upstream repo, since some BSC code has already been merged there, and last time I checked there was a pending PR for BSC validator support), or invite the ankr people to come and comment on our issues here.

**Comment by g4bwy** [comment]
I just asked it [here](https://github.com/ledgerwatch/erigon/issues/3219)

**Comment by koen84** [comment]
@scream7 Currently 4954244775936 bytes (4.6 TiB) at block 14558300

**Comment by scream7** [comment]
> @scream7 Currently 4954244775936 bytes (4.6 TiB) at block 14558300

thanks, just used around 1.2T by default prune parameter( almost done block execution stage)

**Comment by nikolinsko** [comment]
fully synced with hrtc set to 10000 -> 1.4T
